basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/UndertowOperations.java

```diff
@@ -201,7 +201,7 @@ public String createSocketBinding() throws CommandFailedException, IOException {
         String name = "UndertowSocketBinding_" + RandomStringUtils.randomAlphanumeric(6);
         try (OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient()) {
             client.apply(new AddSocketBinding.Builder(name)
-                    .port(ThreadLocalRandom.current().nextInt(10000, 19999))
+                    .port(AvailablePortFinder.getNextAvailable())
```

Comment:
It seems to give ephemeral port numbers since it uses new java.net.ServerSocket(0) so should be OK but not guaranteed. It might be good idea to log selected port number. WDYT?